update /new page with new copy for all languages (fix #16333)

wen-2018 opened the pull request:
_If this changeset needs to go into the FXC codebase, please add the `WMO and FXC` label._

## One-line summary

This updates updates the /firefox/new with the new copy for all languages

## Significant changes and points to review

- Check firefox/new content for all languages

## Issue / Bugzilla link

https://github.com/mozilla/bedrock/issues/16333

## Testing

http://localhost:8000/en-US/firefox/new/
http://localhost:8000/en-CA/firefox/new/
http://localhost:8000/en-GB/firefox/new/
http://localhost:8000/fr/firefox/new/
http://localhost:8000/de/firefox/new/

wen-2018 commented on bedrock/firefox/templates/firefox/new/desktop/firefox-new-refresh.html:
TODO: update `See Release Notes` translations

janbrasna commented on bedrock/firefox/templates/firefox/new/desktop/firefox-new-refresh-all.html:
```diff
@@ -0,0 +1,267 @@
+{% macro default_browser_checkbox(id='default-browser-checkbox') -%}
+<label for="{{ id }}" class="default-browser-label hidden">
+  <input type="checkbox" id="{{ id }}" class="default-browser-checkbox">
+  Set Firefox as your default browser.
```
hardcoded EN

janbrasna commented on bedrock/firefox/templates/firefox/new/desktop/firefox-new-refresh-all.html:
```diff
@@ -0,0 +1,267 @@
+        },
+        optional_attributes={
+          'class': 'c-block-media-img',
+          'alt': 'Firefox on desktop and mobile.',
```
do we translate all alts too?

janbrasna commented on bedrock/firefox/templates/firefox/new/desktop/firefox-new-refresh-all.html:
```diff
@@ -0,0 +1,267 @@
+
+<section class="mzp-l-content t-releases">
+  <div class="mzp-c-emphasis-box js-animate">
+    <h2 class="mzp-c-section-heading">Latest Firefox features</h2>
```
hardcoded heading

janbrasna commented on bedrock/firefox/templates/firefox/new/desktop/firefox-new-refresh-all.html:
```diff
@@ -0,0 +1,267 @@
+        <p>{{ ftl('firefox-new-refresh-block-ads') }}</p>
+      </li>
+      <li class="t-cursor">
+        <img alt="Multiple cursors." src="{{ static('img/firefox/new/desktop/firefox-new-refresh/multi.svg') }}" width="134" height="38">
```
EN alt

janbrasna commented on bedrock/firefox/templates/firefox/new/desktop/firefox-new-refresh-all.html:
```diff
@@ -0,0 +1,267 @@
+        <p>{{ ftl('firefox-new-refresh-go-into-reading') }}</p>
+      </li>
+      <li>
+        <img alt="Lots of open tabs." src="{{ static('img/firefox/new/desktop/firefox-new-refresh/tabs.svg') }}" width="72" height="54">
```
EN alt

janbrasna commented on bedrock/firefox/templates/firefox/new/desktop/firefox-new-refresh-all.html:
```diff
@@ -0,0 +1,267 @@
+          <button id="protection-report" type="button" class="mzp-c-cta-link">{{ ftl('firefox-desktop-download-see-your-report') }}</button>
+        </div>
+        <div class="c-block-media l-v-end l-h-end l-media-constrain-on-sm">
+        <img alt="Shield and hand protecting a browser tab." class="c-block-media-img" src="{{ static('img/firefox/new/desktop/firefox-new-refresh/block.svg') }}" width="220">
```
EN alt

janbrasna commented on bedrock/firefox/templates/firefox/new/desktop/firefox-new-refresh-all.html:
```diff
@@ -0,0 +1,267 @@
+          <p>{{ ftl('firefox-new-refresh-stay-focused', tomato_clock_url='https://addons.mozilla.org/firefox/addon/tomato-clock/', turn_off_the_lights_url='https://addons.mozilla.org/firefox/addon/turn-off-the-lights/') }}</p>
+        </div>
+        <div class="c-block-media l-v-center l-h-start l-media-constrain-on-sm">
+        <img alt="A human brain connected to the world." class="c-block-media-img" src="{{ static('img/firefox/new/desktop/firefox-new-refresh/think.svg') }}" width="260">
```
EN alt

janbrasna commented on bedrock/firefox/templates/firefox/new/desktop/firefox-new-refresh-all.html:
```diff
@@ -0,0 +1,267 @@
+          </div>
+        </div>
+        <div class="c-block-media l-v-end l-h-end l-media-constrain-on-sm">
+        <img alt="Desktop, laptop, and phone. " class="c-block-media-img" src="{{ static('img/firefox/new/desktop/firefox-new-refresh/screens.svg') }}" width="350">
```
EN alt

janbrasna commented on bedrock/firefox/templates/firefox/new/desktop/firefox-new-refresh-all.html:
```diff
@@ -0,0 +1,267 @@
+
+    <div class="c-screen">
+      <div class="c-screenshot">
+        <img alt="Firefox in dark mode." src="{{ static('img/firefox/new/desktop/firefox-new-refresh/dark.svg') }}" width="1000">
```
EN alt

janbrasna commented on bedrock/firefox/templates/firefox/new/desktop/firefox-new-refresh-all.html:
```diff
@@ -0,0 +1,267 @@
+  <div class="mzp-c-emphasis-box js-animate">
+    <div class="c-free">
+      <div class="c-free-img">
+        <img alt="Firefox on a desktop." src="{{ static('img/firefox/new/desktop/firefox-new-refresh/free.svg') }}" width="300">
```
EN alt

janbrasna commented on bedrock/firefox/templates/firefox/new/desktop/firefox-new-refresh-all.html:
```diff
@@ -0,0 +1,267 @@
+  </section>
+
+  <section class="c-support">
+    {% set questions_attrs = 'href="https://support.mozilla.org/products/firefox%s&utm_content=mozilla-support" rel="external noopener" data-cta-text="Mozilla support"'|safe|format(referrals) %}
```
I'm not entirely sure here, but I think they _do_ translate the "Mozilla support" term.

janbrasna commented on bedrock/firefox/templates/firefox/new/desktop/firefox-new-refresh-all.html:
```diff
@@ -0,0 +1,267 @@
+{% extends "firefox/new/desktop/base.html" %}
+
+{% set referrals = '?utm_source=www.mozilla.org&utm_medium=referral&utm_campaign=firefox-desktop' %}
+{% set ios_url = app_store_url('firefox', 'mozilla-org-firefox-new') %}
```
i see source/campaign/content-like ct added to ios_url but empty for google play store button -- does it know how to pick up some? the play store uses utm_* that might be able to also be set?

janbrasna reviewed (commented):
I have checked with l10n and separating templates to support pre-existing translations doesn't make sense — as anything that hits the l10n repo will trigger the agency translations for fr+de anyways with no way to exclude it. So the word is to scrap the existing translations for now, and let l10n pipelines do their job. (And once it arrives back, maybe just verify it's not /that much/ different, or verify with the team that briefed these experiment translations that whatever comes back from smartling is acceptable.)

I know this might be preliminary, but just when I noticed I'll point out some English occurrences:

wen-2018 commented:
I have updated the PR to use one template for this page and added translations for iamge alts. Currently, the `en-CA` is using the `en-GB` content instead of the `en-US` content

janbrasna commented on bedrock/firefox/views.py:
```diff
@@ -826,17 +826,13 @@ def get_context_data(self, **kwargs):
-                template = "firefox/new/desktop/firefox-new-refresh.html"
-            else:
-                template = "firefox/new/desktop/download.html"
+            template = "firefox/new/desktop/firefox-new-refresh.html"
```
Hmh actually is the `ftl_file_is_active("firefox/new/desktop")` condition above this right? Wouldn’t the check be for …/new/refresh* at this point?

janbrasna reviewed (commented):
Noticed something in activation logic, so I’m better asking:

stephaniehobson commented:
Closing in favour of https://github.com/mozmeao/springfield/pull/351

@janbrasna I believe I addressed your review feedback with the port as well 🤞
